Visibility Calculator does not work if `begin`/`end` does not match step size

### Description

`EphemerisVisibility` crashes if the `begin` and `end` values are not rounded to the value of `step_size`. `step_size` is typcially 60s, and `EphemerisVisibility` will round the input value of `begin` and `end` to the value of `step_size`, as it does not make sense for a calculation at a 1 minute resolution to start at for example, 1 minute and 10.1234 seconds. 

However, `Ephemeris` does not perform this rounding, leading to a mismatch. This causes Visibility calculations to crash as the timestamps in the ephemeris do not match the `begin` and `end` values.

### Steps to Reproduce

1. Check out the visibility calculator endpoint in `across-server` https://github.com/ACROSS-Team/across-server/tree/73-feature-add-get-toolsvisibility-calculatorwindows-endpoint
2. Run the API, go to swagger UI and go to the visibility calculator.
3. Do something similar to the following:

<img width="692" height="751" alt="Image" src="https://github.com/user-attachments/assets/946f7895-8161-46bd-b8fd-ac1e6822f2b4" />

4. Observe 500: Internal Server error.
5. Observer server error `AssertionError: Time outside of ephemeris of range`




### Expected Behavior

It should work.

--- a/across/tools/ephemeris/base.py
+++ b/across/tools/ephemeris/base.py
@@ -116,9 +116,10 @@ class Ephemeris(ABC):
         elif isinstance(step_size, (int, float)):
             self.step_size = TimeDelta(step_size * u.s)
 
-        # Round the end time to the nearest step size
-        steps = np.ceil(self.end.unix / self.step_size.to_value(u.s))
-        self.end = Time(steps * self.step_size.to_value(u.s), format="unix")
+        # Align begin/end to step grid (floor)
+        s = self.step_size.to_value(u.s)
+        self.begin = Time((self.begin.unix // s) * s, format="unix")
+        self.end = Time((self.end.unix // s) * s, format="unix")
 
         # Compute range of timestamps
         self.timestamp = self._compute_timestamp()

